cachito/workers/scm.py

```diff
@@ -153,14 +153,24 @@ def clone_and_archive(self, gitsubmodule=False):
         with tempfile.TemporaryDirectory(prefix="cachito-") as temp_dir:
             log.debug("Cloning the Git repository from %s", self.url)
             clone_path = os.path.join(temp_dir, "repo")
+
+            kwargs = {
+                "no_checkout": True,
+                # Don't allow git to prompt for a username if we don't have access
+                "env": {"GIT_TERMINAL_PROMPT": "0"},
+            }
+
+            if include_git_dir:
+                # In case we don't need to keep the Git history, clone with depth 1 to save
+                # resources on large repos
+                kwargs["depth"] = 1
+
             try:
-                repo = git.repo.Repo.clone_from(
-                    self.url,
-                    clone_path,
-                    no_checkout=True,
-                    # Don't allow git to prompt for a username if we don't have access
-                    env={"GIT_TERMINAL_PROMPT": "0"},
-                )
+                repo = git.repo.Repo.clone_from(self.url, clone_path, **kwargs)
+
+                if include_git_dir:
```

Comment:
And what if we need a special commit and it's not the last one (and `include_git_dir` is set to False)?

Reply:
If `include_git_dir` is set to `False`, a deep clone will be made, which means all commits will already be there. This is the current behavior, by the way.